yeat/cli/config.py

```diff
@@ -130,18 +140,24 @@ def to_dict(self, args, readtype="all"):
         else:
             samples = self.batch[readtype]["samples"]
             assemblers = self.batch[readtype]["assemblers"]
+        label_to_samples = {}
+        for assembler in assemblers:
+            label_to_samples[assembler.label] = [
+                sample for sample in assembler.samples if sample in samples
+            ]
         return dict(
             samples={label: sample.to_string() for label, sample in samples.items()},
             labels=[assembler.label for assembler in assemblers],
             assemblers={assembler.label: assembler.algorithm for assembler in assemblers},
             extra_args={assembler.label: assembler.extra_args for assembler in assemblers},
-            label_to_samples={assembler.label: assembler.samples for assembler in assemblers},
+            label_to_samples=label_to_samples,
             sample_readtype={label: sample.readtype for label, sample in samples.items()},
             threads=args.threads,
             downsample=args.downsample,
             coverage=args.coverage,
             genomesize=args.genome_size,
             seed=args.seed,
+            length_required=args.length_required,
```

Comment:
`length_required` is introduced here for the snakefiles to use during the `fastp` rules.